test/k8s-integration/cluster.go

```diff
@@ -379,3 +394,10 @@ func getKubeClient() (kubernetes.Interface, error) {
 	}
 	return kubeClient, nil
 }
+
+func isGKEDeploymentInstalledByDefault(clusterVersion string) bool {
```

Comment:
Is it possible to query the cluster to see if it's already enabled, instead of encoding versions?

Reply:
We could parse the output of `gcloud beta container clusters describe`. It might be good to test these version bounds for defaulting tho, and it's also a teeny bit less work

Reply:
This function should check if we detect a pdcsi-node daemonset in kube-system namespace.(as this indirectly indicates that the component is enabled)

Reply:
I think it's still good to use version bounds so that defaulting is only done in the versions we expect, but not a huge benefit since other test suites already cover defaulting testing. No strong preference either way

Calling kubectl to get DaemonSet info also requires parsing, in which case I would prefer getting the `addonsConfig` from `clusters describe` which is the definitive state of whether GKE deployment should be running.

Reply:
Agree that checking addonsConfig is the right way to do it.
The only issue with checking k8s version, is that we would need to revisit and update the logic again, if there is a on-by-default is enabled for 1.17 clusters for e.g and also then it may get complicated with all the patch version comparision.
For immediate relief to the test failures I am ok with the k8s version check, but eventually we should check addonsConfig.

WDYT @msau42 ?

Reply:
Long term I agree checking addons config is the right way. This is fine for now

Reply:
Ah right yeah all the backports. Will add a TODO here.

Reply:
Actually once the GKE go client refactor happens, we could just always explicitly enable/disable on cluster create. Defaulting logic is already tested elsewhere so not having coverage here is OK.